vllm/distributed/kv_events.py

```diff
@@ -44,16 +45,16 @@ class KVCacheEvent(
 
 
 class BlockStored(KVCacheEvent):
-    block_hashes: list[int]
-    parent_block_hash: Optional[int]
+    block_hashes: list[BlockHash]
+    parent_block_hash: Optional[BlockHash]
```

Comment:
I think we should change these ones from `BlockHash` to `Union[BlockHash, int]`, and have an env variable control which is produced.

In the int case we should do `int.from_bytes(hash, byte_order="big) & ((1 << 64) - 1)` to convert (regardless of which hash function is configured).

This way we can avoid breaking existing consumers of the events. Technically this will change the behaviour in the sha256 case since we'll now return 64bit rather than 256bit ints, but I think this is ok.

TBD on the default value for that env var. Probably it should be int and then we can change to bytes in a major version update.